### DISCUSSION TITLE
Error while run tests after updates

### DISCUSSION THREAD

**Discussion by hchen139** [idaholab/moose#28340] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

After updating conda moose environment and get the latest updates for moose, I got the following error message while trying to run_test.

<frozen importlib._bootstrap>:241: RuntimeWarning: compile time version 3.10 of module 'hit' does not match runtime version 3.11           
Segmentation fault: 11

Any help on having the issue solved will be much appreciated.

**Comment by GiudGiud** [comment]
Hello

The hit warning could be red herring

Let's run a single input for example in test/tests/kernels/simple_diffusion and see if the error can be isolated there

**Reply by hchen139** [reply]
I run the test as
../../../moose_test-opt -i simple_diffusion.i
and got the following error
Illegal instruction: 4

BTW, I'm running MacOS Sonoma version14.6.1 (23G93).

**Reply by GiudGiud** [reply]
ugh. This does not give us much 

Is this a fresh install? If not then I would git clean -xfD the repo (in moose/) then try again
If so, then we ll need to run a debugger and hope we get better information
https://mooseframework.inl.gov/application_development/debugging.html

**Reply by milljm** [reply]
I wonder if this could be folks performing an update with Conda, getting the OpenMPI compilers by accident, and running MOOSE which has some stale libs built with MPICH.

I think what @GiudGiud mentioned is the right approach. Clean the build, and build anew. But let's also make sure that you are using the 'better' stack. Can you perform the following and report back:

```bash
conda activate moose
condo list | grep moose
```

**Reply by hchen139** [reply]
I did rebuild after clean -xdf, even reinstalled moose and conda. The issue now changed to all tests failed and when ran the simple_diffusion.i test, got the same error. Here is what returned after running 'conda list | grep moose':
moose-dev                 2024.07.27                mpich    https://conda.software.inl.gov/public
moose-libmesh             2024.07.27              mpich_0    https://conda.software.inl.gov/public
moose-libmesh-vtk         9.3.0                   mpich_0    https://conda.software.inl.gov/public
moose-mpi                 2024.07.10                mpich    https://conda.software.inl.gov/public
moose-peacock             2024.07.10              mpich_0    https://conda.software.inl.gov/public
moose-petsc               3.20.3                  mpich_1    https://conda.software.inl.gov/public
moose-tools               2024.07.19      py311h09a46a1_0    https://conda.software.inl.gov/public
moose-wasp                2024.05.08              build_0    https://conda.software.inl.gov/public

**Reply by milljm** [reply]
I think this qualifies for running the diagnostic script in its more "troubleshooting" fashion. To see if this machine is having other more serious issues, or if this is an environment issue.

```bash
cd ~/projects/moose/scripts
export METHOD=opt METHODS=opt MOOSE_JOBS=6
./diagnostics.sh --full-build
```

Diagnostics will attempt to do _everything_ in a clean environment, all in a temp directory, from scratch. So it may take a while. But it will help tell us if building and running moose is even possible on this machine.

Edit:
I am running this on my own, as it's been a while... and I seem to hit a snag when building libMesh. Might happen to you. Working on a fix.

Edit:
If you are running into an issue with the above (failing to build libMesh) try modifying the `moose/scripts/diagnostics.sh` script like so:

at line (241) https://github.com/idaholab/moose/blob/e0de6598545e9ba8ea902d1ef1ef2560b05decc2/scripts/diagnostics.sh#L241
```diff
-MOOSE_PACKAGES="${MPI_PACKAGES} packaging pyaml jinja2 python"
+MOOSE_PACKAGES="${MPI_PACKAGES} zlib packaging pyaml jinja2 python"
```

(add `zlib` to the list)

**Reply by hchen139** [reply]
I can run the diagnostics without any changes to the file. But it completed with 7 tests failed, all of which are for the web_server_control:

controls/web_server_control.get_postprocessor ................................................. FAILED (CRASH)
controls/web_server_control.connect_port ...................................................... FAILED (CRASH)
controls/web_server_control.errors/set_controllable_no_exist ................. FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/postprocessor_no_exist .................... FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_unregistered_type ........ FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_bad_convert_json ......... FAILED (EXPECTED ERROR MISSING)
controls/web_server_control.errors/set_controllable_vector_non_array ......... FAILED (EXPECTED ERROR MISSING)

**Reply by milljm** [reply]
Yeah that's normal (the web server tests failing).

Okay, looks like this means there is going to be _something_ in your environment causing the failures.

We will need to take a closer look at the difference between these two environments:

Open a new terminal and:
```bash
conda activate moose
env | sort > ~/bad_env
```

And this environment
```bash
/usr/bin/env -i bash --norc --noprofile
# do the bits to manually source the conda environment. Just a guess if you installed
# Miniforge as described on our wiki to the letter:
export PATH=~/miniforge/bin:$PATH
source activate ~/miniforge/envs/moose
env | sort > ~/good_env
```

And then compare the two:
```bash
sdiff ~/good_env ~/bad_env
```

**Reply by hchen139** [reply]
There is no miniforge directory on my computer, and only miniconda3.

**Reply by milljm** [reply]
In your case then:
```bash
/usr/bin/env -i bash --norc --noprofile
export PATH=~/miniconda3/bin:$PATH
source activate ~/miniconda3/envs/moose
env | sort > ~/good_env
```

**Reply by hchen139** [reply]
This is the comparison result:
<img width="1170" alt="Screenshot 2024-08-12 at 1 41 30 PM" src="https://github.com/user-attachments/assets/6657ade8-3825-4c75-bca1-ce874fede1ce">
<img width="1152" alt="Screenshot 2024-08-12 at 1 41 52 PM" src="https://github.com/user-attachments/assets/218198eb-1b71-4beb-9a79-01d00bcb642a">

**Reply by milljm** [reply]
Much of that looks fine to me.

I did forget to ask if you can run `moose_test-opt`, while in the 'good_env':

```bash
/usr/bin/env -i bash --norc --noprofile
export PATH=~/miniconda3/bin:$PATH
source activate ~/miniconda3/envs/moose
```
And then do what you did when you:
>../../../moose_test-opt -I simple_diffusion.i

If you can't, then as strange as it sounds because it seems you were asked, and performed a clean, your current copy of MOOSE is simply not being built correctly/sanely (Still dirty. e.g a build using old no-longer-existing-libraries after a `conda update` etc).


Edit:
`PATH` is different, and truncated. So if the above fails to yield results, we'll investigate that next (I'll need a comparison to the two).

**Reply by hchen139** [reply]
Following the above instruction, I got the same error message: Illegal instruction: 4

**Reply by milljm** [reply]
I think this may have something to do with a recent libMesh update. We have been discussing this off-discussions, and it appears the issue goes away by using a slightly older `moose-libmesh` package. This is super not ideal, but can you try the following:

```bash
conda deactivate
conda activate base
conda create -n oldermoose moose-dev=2024.07.19
conda activate oldermoose
```

And then of course, clean your build (@MinMorsh hint, this is what _you_ need to do to solve your VTK issue):
```bash
cd ~/projects/moose
git clean -xfd; git submodule foreach --recursive git clean -xfd
cd test
make
./run_tests
```
Save your work before running the `git clean` commands, if you have any changes... Please report if this prevents the `Illegal instruction` issue. This is NOT a solution. I am just collecting more data points.

**Comment by MinMorsh** [comment]
I ran through the same issue when I updated my moose and conda. Initially, I was getting the segmentation fault. Then I tried updating running `./update_and_rebuild_petsc.sh`, `./update_and_rebuild_libmesh.sh` and `update_and_rebuild_wasp.sh` which led me to this error while running `./run_tests`: [Logfile.txt](https://github.com/user-attachments/files/16684199/Logfile.txt). Seems like libMesh could not locate the VTK. 

@hchen139 Could you find any solution?

**Reply by milljm** [reply]
@MinMorsh, your issue not related to `Illegal instruction: 4`. Please begin a new discussion (if you already have, I apologize... I am simply addressing discussions as I receive them in my email). Your issue is a simple one as well, which I cannot answer here (cannot mark my post as answering this thread as it does not pertain to the original post).

**Comment by roystgnr** [comment]
I think I have a possible explanation for this, with a fix in https://github.com/libMesh/libmesh/pull/3940

I'll post a further update once the fix is downstream and we have conda modules with it ready for testing.

**Comment by GiudGiud** [comment, marked as answer]
The fix is now in all branches of moose. If you update with 
```
conda install moose-dev=2024.08.30 mpich
```
then:
- save all your work in moose
- go to moose/
- `git clean -Xfd`
- then recompile with make
you should get working binaries

**Reply by hchen139** [reply]
Problem solved! Much appreciated for the help to the team.
